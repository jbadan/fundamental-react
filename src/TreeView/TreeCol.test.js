import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import TreeView from './TreeView';

describe('<TreeView.Col />', () => {
    const treeCol = (
        <TreeView.Col>Column Header 1</TreeView.Col>
    );

    test('create tree component', () => {
        // multi-level tree
        let component = renderer.create(treeCol);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the TreeCol component', () => {
            const element = mount(
                <TreeView.Col
                    data-sample='Sample' />);

            expect(
                element.find('.fd-tree__col').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
