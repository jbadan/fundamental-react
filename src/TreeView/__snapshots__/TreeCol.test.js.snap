// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeView.Col /> create tree component 1`] = `
<div
  className="fd-tree__col"
>
  Column Header 1
</div>
`;
